### ISSUE TITLE
How can I store a part number and then keep adding part numbers into an array using the same select lists?  (add and appending an array in Sprig)

### ISSUE THREAD

**Issue by terryupton** [putyourlightson/craft-sprig#382]
### Support Request

I am building an enquiry form with part code generator and I am a little stuck on one aspect.
The screenshot hopefully gives some context.
So the user will go on and be able to choose options from the select lists to build their part number.
Once all options selected I want them to click the button to add the part code and then repeat the process using the same select boxes for the next part code. 

I have all of this working except the ability to store the part number on the button click and then push the next part number into the array. I am just a little stuck on the logic and handling this via Sprig.

I have looked at the Sprig form and it is kind of similar to the table but not quite as I don't really want to repeat all the select list multiple times. I just want to append an input field with another part number and rinse and repeat.

**First**
<img width="1555" alt="Screenshot 2024-07-25 at 20 13 01" src="https://github.com/user-attachments/assets/2dd99199-631e-461d-b00a-98dc94e4b42a">

**After Adding the first part number:**
The options reset and you can build the next part.
<img width="1586" alt="Screenshot 2024-07-25 at 20 36 01" src="https://github.com/user-attachments/assets/6451b604-1985-4023-85a7-1cbb5c7ef827">

**After Adding another (second) part number:**
<img width="1584" alt="Screenshot 2024-07-25 at 20 38 00" src="https://github.com/user-attachments/assets/9cecda49-d4b1-4742-814f-b458bf3da7a9">

And so on....

Can you direct me as to how I can store the new part number and then append the part numbers each time I click to add a new one?  Thanks Ben 🙏🏻

Thanks
Terry

### Plugin Version

Latest Version

**Comment by bencroker** [comment]
Something like this, perhaps?

```twig
{% set partCodes = (partCodes ?? null) ? partCodes|split(',') : [] %}

Part Codes:
{% for partCode in partCodes %}
    {{ hiddenInput('partCodes[]', partCode) }}
    {{ partCode }}
{% endfor %}

{% set newPartCode = 'XYZ' %}

<button sprig s-val:part-codes="{{ partCodes|push(newPartCode)|join(',') }}">
    Add Product
</button>
```

**Comment by terryupton** [comment]
Thanks very much Ben.
I got somewhere similar in my logic, before I stopped and reached out. 
I keep hitting a similar issue as per the code you posted above and that it works fine for the first part code , but when you come to do the next part code, I am seeing this.

```
TypeError
explode(): Argument #2 ($string) must be of type string, array given
```

<img width="1937" alt="Screenshot 2024-07-26 at 10 18 38" src="https://github.com/user-attachments/assets/1da96a54-01c7-4940-a518-24b73c71c016">

If I change `{{ hiddenInput('partCodes[]', partCode) }}` to not an array `{{ hiddenInput('partCodes', partCode) }}` which fixes then error, then it works for the next code, but then it clears the first code when you start adding the 3rd code (as in it only keeps the last one, which makes sense as its not an array).

Thank you.
